### DISCUSSION TITLE
How to get global step from checkpoint?

### DISCUSSION THREAD

**Discussion by timothybrooks** [Lightning-AI/pytorch-lightning#6463] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hello--

I am saving checkpoints inside my module using `self.trainer.save_checkpoint(path)`. I am able to load these checkpoints into the model using `MyModel.load_from_checkpoint(path)` and trainer using `Trainer(resume_from_checkpoint=path)`. However, both the resulting model and trainer have `global_step=0` regardless of the step when saving. From the documentation I was of the impression that checkpoints saved the global step, which is important for my use case. How can I attain the global step from a checkpoint?

**Comment by carmocca** [comment, marked as answer]
Discussion continued in https://github.com/PyTorchLightning/pytorch-lightning/issues/6470
